view-user/src/utils/api_methods.ts

```diff
@@ -30,6 +41,43 @@ export async function getBingoNumber(): Promise<BingoNumber[]> {
   }
 }
 
+// websocket通信でBingoNumberを取得
+export async function subscriptionBingoNumber(): Promise<BingoNumber[]> {
+  try {
+    const response = await client.subscribe({
+      query: gql`
+        subscription MySubscription {
+          bingo_number {
+            data
+            id
+          }
+        }
+      `,
+    });
+    // response は Observable なのでそのままでは返せない
+    // 適切な方法で Observable を処理してデータを取得する必要がある
+
+    // 例: Observable を Promise に変換して処理する
+    return new Promise<BingoNumber[]>((resolve, reject) => {
+      response.subscribe({
+        next(data) {
+          // データの処理
+          // ここでは data.data.bingo_number を使ってデータを取得
+          const bingoNumbers: BingoNumber[] = data.data.bingo_number;
+          resolve(bingoNumbers);
+        },
+        error(error) {
+          console.error('Subscription error:', error);
+          reject(error);
+        },
+      });
+    });
+  } catch (error) {
+    console.error('Error fetching data:', error);
+    return [];
```

Comment:
```suggestion
    return new Promise<BingoNumber[]>((resolve, reject) => {
      response.subscribe({
        next: data => resolve(data.data.bingo_number),
        error: error => {
          console.error('Subscription error:', error);
          reject(error);
        },
      });
    });
  } catch (error) {
    console.error('Error fetching data:', error);
    return [];
  }
}
```
簡略化だけしました。
正直私もよく分かりません。

Reply:
以下、chatgpt
resolve と reject は、Promiseの状態を操作するための関数です。Promiseは非同期操作の結果を表すオブジェクトで、成功した場合にはresolveが呼ばれ、失敗した場合にはrejectが呼ばれます。

具体的には以下のように使われます：

resolve(value): Promiseが成功したことを示し、引数 value が解決値として渡されます。この解決値はPromiseを待っている .then() メソッドに渡されます。例えば、resolve(42) のように使うことで、非同期操作が成功し、値 42 が関連する .then() メソッドに渡されます。

reject(reason): Promiseが失敗したことを示し、引数 reason が失敗の理由を示す情報として渡されます。この理由は、Promiseを待っている .catch() メソッドに渡されます。例えば、reject(new Error('Something went wrong')) のように使うことで、非同期操作が失敗し、新しいエラーオブジェクトが関連する .catch() メソッドに渡されます。

コードサンプルの中では、resolve(bingoNumbers) が成功時に解決値として bingoNumbers（BingoNumberの配列）を返し、reject(error) が失敗時にエラー情報を渡しています。これによって、response.subscribe() 内の非同期処理が終了した後の処理を制御し、外部からPromiseの状態を操作しています。